sky/optimizer.py

```diff
@@ -120,6 +120,9 @@ def optimize(dag: 'dag_lib.Dag',
                 for a task.
             exceptions.NoCloudAccessError: if no public clouds are enabled.
         """
+        for task in dag.tasks:
+            _check_specified_clouds(task)
```

Comment:
It seems like there is a `sky check` in the `_check_specified_clouds` function, so it will be called the number of tasks times in one dag pipeline. Should we only do one sky check for a dag?

Reply:
This is now fixed. Thanks for catching this!